controllers/operator/routes.go

```diff
@@ -285,7 +287,7 @@ func (r *AuthenticationReconciler) reconcileRoute(ctx context.Context, instance
 			// Requeue after creating new route
 			*needToRequeue = true
 		}
-	} else {
+	} else if shouldHaveRoutes(instance) {
```

Comment:
Should this be checked before the route get above is done?  If the flag is on, the get above will throw an error and shouldn't.

Reply:
That ends up not being an issue, but something _later on_ does. Fixing that.

Reply:
Also going to clean this up a little bit given it took multiple passes for the issue to be seen.